onboard/merge_iterator.go

```diff
@@ -0,0 +1,170 @@
+package onboard
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/treeverse/lakefs/catalog/rocks"
+)
+
+// prefixMergeIterator takes inventory iterator and commit iterator and merges them.
+// If an entry's prefix is in `prefixes`, it would be taken from invIt.
+// No entry from committedIt would be returned if its prefix is in prefixes.
+type prefixMergeIterator struct {
```

Comment:
I think this iterator needs a test.  Doesn't have to be in this PR, but should be before the upcoming release.

Reply:
I will add it as part of this PR.